From e1e8d8f982bf1b05fa81ca77fd7af6c5587d8d10 Mon Sep 17 00:00:00 2001
From: Sugar Zhang <sugar.zhang@rock-chips.com>
Date: Wed, 2 Sep 2015 16:46:36 +0800
Subject: [PATCH] ASoC: rockchip: spdif: increase dma burst size

increase dma burst size from 4 to 16 to optimize dma transfer.

Change-Id: I1a9112e6b89ed1f29d24a3e4b5f42bf4201a9e71
Signed-off-by: Sugar Zhang <sugar.zhang@rock-chips.com>
---
 sound/soc/rockchip/rk_spdif.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/sound/soc/rockchip/rk_spdif.c b/sound/soc/rockchip/rk_spdif.c
index 003569a9ff74..12f5f52a6fed 100755
--- a/sound/soc/rockchip/rk_spdif.c
+++ b/sound/soc/rockchip/rk_spdif.c
@@ -135,6 +135,9 @@
 #define INTCR_SDBEIE_ENABLE		(0x1 << 4)
 #define INTCR_SDBEIE_MASK		(0x1 << 4)
 
+/* size * width: 16*4 = 64 bytes */
+#define SPDIF_DMA_BURST_SIZE		(16)
+
 struct rockchip_spdif_info {
 	spinlock_t lock;/*lock parmeter setting.*/
 	void __iomem *regs;
@@ -451,7 +454,7 @@ static int spdif_probe(struct platform_device *pdev)
 
 	spdif->dma_playback.addr = mem_res->start + SMPDR;
 	spdif->dma_playback.addr_width = DMA_SLAVE_BUSWIDTH_4_BYTES;
-	spdif->dma_playback.maxburst = 4;
+	spdif->dma_playback.maxburst = SPDIF_DMA_BURST_SIZE;
 
 	ret = snd_soc_register_component(&pdev->dev,
 					 &rockchip_spdif_component,
-- 
2.35.3

